### ISSUE TITLE
slurm: collect a balanced oracle-imitation training dataset for #1496

### ISSUE THREAD

**Issue by ll7** [ll7/robot_sf_ll7#6127]
## Current issue-planning review — 2026-09-15

**Audit disposition:** `blocked`. Active open PR evidence is present; the issue remains open and no ownership change is made.

**First action:** Reuse #1470/#1397 leakage and split policy; freeze family/action strata before collection.

### Ordered implementation / investigation plan

1. Reuse #1470/#1397 leakage and split policy; freeze family/action strata before collection.
2. Count existing eligible data first; six episodes/616steps are loader-smoke only, not a comparison dataset.
3. Prepare one bounded collection packet targeting at least10,000 usable transitions and10 nondegenerate episodes per selected stratum, retaining invalid/fallback row provenance.
4. Run through authorized private ops only after input/dry-run checks; verify durable checksums and a BC loader smoke, then hand the exact manifest to #1496.

### Files and owners to inspect

- expert_traj_v1 collection path
- train/validation/holdout manifest
- #1496 dataset registry

### Acceptance and validation

- Usable counts exclude one-step, invalid, fallback and leakage rows.
- Repeated identical failed fingerprints are not blindly retried.

### Current dependencies and issue-text corrections

- Dependencies: #1470; #1397; #1496; authorized private input/output and compute
- Name literal strata, split hashes, resource cap and expected outputs before ready status.
- Keep collection separate from BC-versus-RL performance claims.
- The supplied draft was checked against the refreshed live issue and complete comment thread; historical rationale is retained below.
- Keep current out-of-scope, ownership, provenance, rights, compute, and claim boundaries intact. Do not close the issue merely because this plan was posted.

### Lifecycle and claim boundary

- Current state labels: `state:running`.
- The issue remains open. No closure, compute submission, release/publication, evidence deletion, or scientific-claim promotion is authorized by this review.
- This review does not request a lifecycle transition; existing assignees, ownership, valid priority labels, and unrelated labels are preserved.
- Proposed non-lifecycle label additions applied with this review: `dependency:has-blockers`.

**Evidence note:** Full live issue body and complete available comment thread were reviewed during the 2026-09-15 refresh. This is planning metadata, not implementation or benchmark evidence; historical measurements remain historical.

## Goal

Collect and durably register the larger balanced oracle-imitation training dataset required before the BC-warm-start-vs-RL-only comparison in #1496. The existing six training episodes / 616 steps are accepted only as loader and overfit-smoke evidence and are too small for a credible comparison.

## Preparation contract

- Reuse the completed #1470/#1397 split and leakage policy and the registered expert_traj_v1 schema.
- Materialize a fixed train/validation/holdout manifest before submission.
- Balance eligible training strata across the selected scenario families and oracle action surface.
- Target at least 10,000 usable training transitions and at least 10 nondegenerate episodes per selected training stratum; the stricter bound wins.
- Exclude one-step, invalid, fallback, and leakage-violating rows from the usable count while retaining their provenance.
- Record exact public main SHA, private input/output URIs, hashes, resource request, timeout, and stop rule in the launch packet.

## Execution and watcher contract

- Submit through robot_sf_ll7-private ops only after the launch packet and dry-run preflight pass.
- Attach the adaptive Slurm watcher: dense polling at launch, wider intervals after stable progress, and exact terminal ingestion.
- Do not automatically repeat an identical failed fingerprint. Retry only after a differential failure reason or a changed packet/environment.
- Promote only compact manifests, hashes, and evidence summaries to the public repository; raw data remains in the durable private result store.

## Acceptance

- Split/leakage preflight passes before submission.
- The durable dataset manifest resolves on an authorized training host and records checksums.
- Usable balance and transition bounds above are met, or the job terminates with a precise evidence-backed insufficiency reason.
- A BC loader smoke consumes the registered dataset without fallback.
- #1496 receives the exact dataset pointer and becomes eligible for its training-comparison launch packet.

## Stop rules

- Stop on leakage, corrupted provenance, unresolved private input, repeated identical failure, or evidence that an eligible stratum cannot meet the minimum without changing the scientific scope.
- Dataset collection does not authorize a performance claim.

Parent: #1496
Predecessors: #1470, #1397, PR #5914, PR #5917, PR #5950.

```yaml
schema: goal_autopilot_preparation.v1
repository: ll7/robot_sf_ll7
issue: 6127
source_body_sha256: 0b1a8e01c448bd2fad5c6d82e3768c287addbe1ef9789a87abe284f9ef38207c
source_comments_sha256: 
audit_schema: open_issue_contract_audit.v1
audit_digest: 1c191411793929640684e0827d763f3a2009a8682ab98dcefa8c2ded3e54f4ea
audit_classification: needs_compute
next_action: route_to_compute_authority
authority: compute_authority
execution_mode: compute
preferred_worker: MaxRunner
expected_pr_runner_label: runner:max
implementation_admitted: False
state_ready_change_proposed: False
mutation_batch: open-issues-20260830-02
```

This packet is preparation evidence only. It never overrides live labels, exact claim state, branch state, typed dependencies, domain gates, compute authority, release authority, or scientific evidence rules.

**Comment by ll7** [comment]
Preparation found one mechanical prerequisite; no owner judgment or Slurm submission is needed yet.

- Pinned public `main`: `0e2036c3e721fc4937a2b63e5838ae180ccaa2bc`.
- The #1397 split/leakage packet, six scenario definitions, non-learning oracle config, and job-13520 private registry/checksums are present.
- The reusable execution command is missing. Job 13520's BC-compatible collector exists only inside its retrieved result tree and hardcodes old commit `9d65072e...` plus the original 12 episode IDs.
- Current tracked `collect_oracle_imitation_candidate_traces.py` does not capture BC observation/action arrays and has no `--preflight` (`exit 2: unrecognized arguments: --preflight`). `collect_expert_trajectories.py` instead expects a PPO checkpoint; its dry-run produces zero actions, so it cannot stand in for the registered non-learning oracle.

I split the deterministic implementation prerequisite into #6128 and marked it `state:ready` / `priority:1`. This issue revives automatically when #6128 merges: private ops will pin the new exact SHA, materialize and hash the full 306-episode manifest, run a no-simulation preflight, and promote one queue row.

The blocked private launch scaffold already fixes the proposed campaign boundary: 300 training episodes (50 per each of six scenario strata, train seeds 1001–1300), the existing 3 validation and 3 holdout rows, at least 10 nondegenerate episodes per training stratum, at least 10,000 usable transitions, two-dimensional action-bin accounting/weights, durable private output, 8 CPU / 96 GiB / 3-hour cap, adaptive watcher cadence `0s → 2s → 10s → 60s → 300s`, and no identical-fingerprint retry.

No job was submitted, cancelled, requeued, or registered.

**Comment by ll7** [comment]
factory-v2 external blocker: {"blocker": {"category": "upstream_dependency", "evidence": ["https://github.com/ll7/robot_sf_ll7/issues/6128"]}, "revival_condition": {"probe": "for child in 6128; do gh api repos/ll7/robot_sf_ll7/issues/$child --jq .state; done", "success_condition": "every named covering child reports CLOSED; re-evaluate parent coverage and residual scope"}}

**Comment by ll7** [comment]
Observed post-submit evidence for job 13665:

- The private Slurm record is terminal success: scheduler state COMPLETED, exit 0:0, elapsed 00:01:55. The adaptive watcher recorded the same terminal result.
- The result bundle was retrieved into the private result store and recorded in private-ops commit 26e3e66. No raw result files or private infrastructure paths are being promoted here.
- Acceptance is not proven. The captured stdout identifies the issue-1397 launch packet and the retrieved manifest schema is oracle-imitation-candidate-traces.v1, not the issue-6127 balanced-dataset manifest.
- The retrieved bundle contains 6 episodes across 3 families. The required balanced_oracle_dataset_manifest.json, expert_traj_v1.npz, SHA-256 inventory, and BC-loader smoke evidence are absent. validate_balanced_oracle_manifest.py fails on the retrieved manifest.
- Therefore #6127 remains open and no dataset pointer, training-readiness claim, promotion, rearm, or resubmission is being made. The exact blocker is a launch-packet/config binding mismatch: the batch script ran its issue-1397 default rather than the queued issue-6127 packet.

**Comment by ll7** [comment]
Observed corrected retry submission (2026-07-24): sanctioned autonomous_queue.py cycle recorded Slurm job 13666 on imech192:a30, currently RUNNING as issue6127-balanced-oracle-retry. scontrol identity: workdir `/home/luttkule/git/robot_sf_ll7.worktrees/issue6127-binding-fix-20260724`, private launcher `auxme/SLURM/issue_1470_oracle_imitation_traces.sl`, 8-hour limit, 1 GPU. Batch stdout proves `PACKET=configs/training/ppo_imitation/oracle_dataset_issue_6127_balanced_launch_packet.yaml` and `RESULTS_ROOT=/home/luttkule/git/robot_sf_ll7.worktrees/issue6127-binding-fix-20260724/output/slurm/issue6127-balanced-oracle-dataset-retry-20260724`; the launcher is dispatching the balanced collector path. This is observed routing/binding evidence only: artifacts are not yet retrieved, balanced-manifest/SHA-256/NPZ/BC-smoke acceptance is not yet proven, and the issue remains open.

**Comment by ll7** [comment]
## Queue routing

This is a SLURM-backed dataset campaign, not a local implementation issue. The complete thread records an active retry job, but acceptance is not yet proven: the balanced manifest, checksums, NPZ, and BC-loader smoke evidence are still missing. It is routed out of `state:ready` and remains `state:running` under the active compute campaign; no local implementation claim is being created.

**Comment by ll7** [comment]
Queue recheck: this issue is already covered by active-SLURM-campaign. Updated the stale implementation-queue label to state:running; no duplicate claim, branch, or worktree was created.

**Comment by ll7** [comment]
## Lifecycle reconciliation — `state:running` was stale for 15 days

This issue has carried **`state:running`** ("External run is currently active") since **2026-07-30T04:21:42Z**. No run has been active behind it. Reconciled today against the scheduler and the private-ops ledger; the label is being corrected to `state:blocked`.

**Evidence**

| Check | Result |
| --- | --- |
| `squeue -u <user>` on the execution host | empty (re-checked twice, 2026-08-14 13:41Z and 13:45Z) |
| Last job for this campaign | job 13666, `FAILED` exit 1:0, ended **2026-07-24**, triaged **2026-08-01** |
| private-ops queue row `issue-6127-balanced-oracle-dataset` | already terminal `state: failed`, `attempts: 2/2` |

So the private-ops queue was honest and the GitHub label was not: the label was applied **six days after the underlying job had already failed**, and outlived it by another fifteen.

**Why it is `state:blocked`, not closed or ready**

The terminal reason is scientific, not infrastructural: the corrected balanced collector (13666) reached its acceptance gate and was rejected because training stratum `classic_doorway_low` produced **1 usable episode against a required minimum of 10**. `attempts` equals `max_attempts`, so no further retry is authorized under the standing no-identical-refire rule.

Per park-don't-close, this issue stays open and revivable. **Revival condition:** a differential change to the collection packet that plausibly fixes the `classic_doorway_low` yield — a changed scenario roster, a relaxed-but-justified stratum minimum, or a longer collection budget — at which point a new bounded campaign can be authorized and the row re-promoted.

**Downstream:** #1496's BC-warm-start-vs-RL-only comparison remains blocked on this dataset. Since compute access ends **2026-09-30**, the yield blocker is worth resolving early rather than late.

Reconciled under ll7/robot_sf_ll7-private-ops#89 (gap G-6, stale lifecycle state withholding capacity).

**Comment by ll7** [comment]
## Implementability routing — 2026-08-19

The campaign is `state:blocked`; `state:ready-to-submit` was stale. The last authorized attempt is terminal, `classic_doorway_low` yielded 1 usable episode against the minimum of 10, and attempts 2/2 are exhausted. Do not submit an unchanged packet.

The bounded local repair is #7604. It will add a per-stratum yield ledger, fail-closed verdict, and exhausted-fingerprint comparison, then prepare the evidence needed to select exactly one differential remedy. It cannot submit compute or change the scenario roster, minimum, budget, eligibility semantics, or source policy.

Revive this campaign only after #7604 completes and a maintainer records one justified differential remedy plus a fresh packet fingerprint and compute authorization.

**Comment by ll7** [comment]
## Open-issue review — 2026-09-02

**Classification:** SLURM data-collection prerequisite for #1496, not a training comparison.

**Frozen quality floor:** reuse the #1470/#1397 split and `expert_traj_v1` schema; materialize train/validation/holdout identities before launch; require at least 10,000 usable transitions and at least 10 nondegenerate episodes per selected stratum, with the stricter condition controlling. Retain but exclude one-step, invalid, fallback, and leakage-violating rows.

**Execution rule:** record public SHA, private inputs/outputs, hashes, resource cap, timeout, and stop rule; submit only through private ops after dry-run; never repeat an identical failed fingerprint without a differential cause.

**Completion:** register a durable, checksum-covered dataset and prove a BC loader consumes it natively, or report precise insufficiency. Only then may #1496 prepare its RL-only versus BC/warm-start comparison. Dataset volume alone establishes no policy-performance claim.

**Comment by ll7** [comment]
## Compute-window execution operator

The current conditional dataset-collection owner is [`ll7/robot_sf_ll7-private-ops#308`](https://github.com/ll7/robot_sf_ll7-private-ops/issues/308).

It must first prove the current #1470/#1397 split, schema, exact selected strata, downstream #1496 loader compatibility, non-duplicate status, full deadline fit, and off-cluster preservation capacity. Only a native disjoint canary pass permits full collection. The minimum quality bounds and holdout separation must not be reduced to finish sooner.

**Comment by ll7** [comment]
## Compute-window operational handoff — 2026-09-10

Conditional private execution is tracked in [`ll7/robot_sf_ll7-private-ops#307`](https://github.com/ll7/robot_sf_ll7-private-ops/issues/307).

The private child must prove that #1496 still needs the dataset, no sufficient duplicate already exists, the #1470/#1397 split and leakage contract remains exact, the oracle/producer/schema path is native, the minimum `10,000` usable transitions and per-stratum episode targets are frozen, and two-copy preservation fits before access loss. It may submit only after all six queue admission predicates pass.

This handoff does not authorize DAgger, change the split/strata/minima, count invalid or leakage-violating rows, or perform training or a scientific claim.
